controllers/imageupdateautomation_controller.go

```diff
@@ -291,6 +291,21 @@ func (r *ImageUpdateAutomationReconciler) SetupWithManager(mgr ctrl.Manager) err
 		Complete(r)
 }
 
+func (r *ImageUpdateAutomationReconciler) patchStatus(ctx context.Context,
+	req ctrl.Request,
+	newStatus imagev1.ImageUpdateAutomationStatus) error {
+
+	var auto imagev1.ImageUpdateAutomation
+	if err := r.Get(ctx, req.NamespacedName, &auto); err != nil {
```

Comment:
This can set the status of a different version of the object than the one we're reconciling. For that reason, I think this is a fragile fix, even if it doesn't cause other problems.